iceoryx_hoofs/test/moduletests/test_posix_semaphore_interface.cpp

```diff
@@ -0,0 +1,312 @@
+    }
+
+    auto result = this->sut->getState();
+    ASSERT_THAT(result.has_error(), Eq(false));
+    EXPECT_THAT(result->value, Eq(0));
+}
+
+
+TYPED_TEST(SemaphoreInterfaceTest, TryWaitAfterPostIsSuccessful)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "22354447-c44d-443c-97a5-f5fdffb09748");
+    ASSERT_FALSE(this->sut->post().has_error());
+    auto call = this->sut->tryWait();
+    ASSERT_THAT(call.has_error(), Eq(false));
+    ASSERT_THAT(*call, Eq(true));
+}
+
+TYPED_TEST(SemaphoreInterfaceTest, TryWaitWithNoPostIsNotSuccessful)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0e5d6817-88a9-4fca-889e-4dbfe2c30e48");
+    ASSERT_FALSE(this->sut->post().has_error());
+    auto call = this->sut->tryWait();
+    ASSERT_THAT(call.has_error(), Eq(false));
+    ASSERT_THAT(*call, Eq(true));
+}
+
+TYPED_TEST(SemaphoreInterfaceTest, WaitValidAfterPostIsNonBlocking)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d4b1de28-89c4-4dfa-a465-8c7cfc652d67");
+    ASSERT_FALSE(this->sut->post().has_error());
+    // this call should not block and should be successful
+    EXPECT_THAT(this->sut->wait().has_error(), Eq(false));
+}
+
+TYPED_TEST(SemaphoreInterfaceTest, WaitBlocksAtLeastDefinedSleepTime)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5869a475-6be3-4b55-aa58-2ebf11d46081");
+    std::atomic<int> counter{0};
+
+    std::thread t1([&] {
+        counter = 1;
+        while (counter.load() != 2)
+        {
+        }
+        std::this_thread::sleep_for(std::chrono::nanoseconds(this->TIMING_TEST_WAIT_TIME.toNanoseconds()));
+        ASSERT_FALSE(this->sut->post().has_error());
+    });
+
+    while (counter.load() != 1)
+    {
+    }
+
+    auto start = std::chrono::steady_clock::now();
+    counter = 2;
+
+    ASSERT_FALSE(this->sut->wait().has_error());
+    auto end = std::chrono::steady_clock::now();
+
+    EXPECT_THAT(std::chrono::nanoseconds(end - start).count(), Ge(this->TIMING_TEST_WAIT_TIME.toNanoseconds()));
+
+    t1.join();
+}
+
+TYPED_TEST(SemaphoreInterfaceTest, TimedWaitBlocksAtLeastDefinedSleepTimeAndSignalsTimeout)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5869a475-6be3-4b55-aa58-2ebf11d46081");
+    std::atomic<int> counter{0};
+
+    std::thread t1([&] {
+        counter = 1;
+        while (counter.load() != 2)
+        {
+        }
+        std::this_thread::sleep_for(std::chrono::nanoseconds(this->TIMING_TEST_WAIT_TIME.toNanoseconds() * 2));
+        ASSERT_FALSE(this->sut->post().has_error());
+    });
+
+    while (counter.load() != 1)
+    {
+    }
+
+    auto start = std::chrono::steady_clock::now();
+    counter = 2;
+
+    auto result = this->sut->timedWait(this->TIMING_TEST_WAIT_TIME);
+    auto end = std::chrono::steady_clock::now();
+
+    ASSERT_FALSE(result.has_error());
+    EXPECT_THAT(*result, Eq(iox::posix::SemaphoreWaitState::TIMEOUT));
+
+    EXPECT_THAT(std::chrono::nanoseconds(end - start).count(), Ge(this->TIMING_TEST_WAIT_TIME.toNanoseconds()));
```

Comment:
Since `timedWait` is tested, we don't need to create another thread calling `post`.